core/core-configuration-layer.el

```diff
@@ -1335,7 +1335,7 @@ to select one."
                  (pkg-dir-name (cdr apkg))
                  (installed-ver
                   (configuration-layer//get-package-version-string pkg))
-                 (elpa-dir (concat user-emacs-directory "elpa/"))
+                 (elpa-dir package-user-dir)
                  (src-dir (expand-file-name
                            (concat rollback-dir (file-name-as-directory
                                                  pkg-dir-name))))
```

Comment:
The default value of package-user-dir does not have trailing slash unlike the code that it is replacing.  Hence this change by itself is not a valid change.  You need to make matching change so that the code works as it did before whether or not package-user-dir has a trailing slash.  For example you can  change expand-file-name on line 1342 like this:

```
(expand-file-name (file-name-as-directory pkg-dir-name) elpa-dir)
```
This works because expand-file-name works whether or not the second argument has trailing slash.